### ISSUE TITLE
stdout.encoding == None

### ISSUE THREAD

**Issue by jeffmal** [cannatag/ldap3#156]
While running nosetests 1.3.7 (or built from master) on a test that required LDAP auth, we get the following error:

```
user.attrs[flask_key] = str(entry[ldap_key])
  File "/opt/searchlight-app/lib/python3.5/site-packages/ldap3/abstract/attribute.py", line 66, in __str__
    return to_stdout_encoding(self.values[0])
  File "/opt/searchlight-app/lib/python3.5/site-packages/ldap3/utils/repr.py", line 42, in to_stdout_encoding
    return value.encode(repr_encoding, errors='backslashreplace').decode(repr_encoding, errors='backslashreplace')
TypeError: encode() argument 1 must be str, not None
```
Could the following lines be added after line 30 of ldap3/utils/repr.py in case stdout.encoding == None:

```
if repr_encoding == None:
        raise
```
Running nosetests with `-s` alleviates the problem by disabling capture, but we were wondering if this would be a better solution.

Thanks,

Jeff

**Comment by cannatag** [comment]
Hi,
wouldn't be better to have:
```
if repr_encoding == None:
        repr_encoding = 'ascii'
```

This would give you a chance to print anyway 

What do you think?

Bye,
Giovanni

**Comment by jeffmal** [comment]
Giovanni,

That works and is probably cleaner.

Thanks,

Jeff

**Comment by cannatag** [comment]
I added:
```
    if not repr_encoding:
        repr_encoding = 'ascii'  # default
```

and will include it in the next release.

Bye,
Giovanni

**Comment by jeffmal** [comment]
That's great.  Thanks, Giovanni.

**Comment by cannatag** [comment]
fixed in version 1.2.released today.
